Add Windows development setup instructions to contributing.rst

paschal533 opened the pull request:
## What was wrong?

Issue #559 

The Contributing section in `docs/contributing.rst` lacked specific instructions
for setting up a development environment on Windows, making it harder for Windows
contributors to get started. Manu and Luca requested Windows-specific guidance to match
the existing Linux and macOS instructions.

## How was it fixed?

Summary of approach:
Added a "Windows Development Setup" subsection under "Setting the stage" in
`docs/contributing.rst`. It includes:
- Installing prerequisites (Python 3.11+, Git, CMake, optional make via Git Bash
  or Chocolatey) with `winget` commands.
- Steps to clone the repo, set up a virtual environment, and install dependencies
  with `pip install -e ".[dev]"`.
- Notes on using PowerShell or Git Bash and verifying with `pytest -v`.
Tested the setup on Windows 10 with Python 3.11.0 to ensure it works.

### To-Do

- [x] Clean up commit history (single commit, already done)
- [x] Add or update documentation related to these changes (added to contributing.rst)
- [x] Add entry to the [release notes](https://github.com/libp2p/py-libp2p/blob/main/newsfragments/README.md)

#### Cute Animal Picture

![a cat](https://cdn.pixabay.com/photo/2024/02/28/07/42/european-shorthair-8601492_640.jpg)

acul71 commented:
@paschal533 
# Diagnosing the reStructuredText Errors

The errors are due to inconsistent indentation in your reStructuredText file. In reST:

- **Directives and code blocks require a blank line after the directive line.**  
  For example, after a line like `.. code:: sh`, you must leave an empty line before starting the indented block.

- **The indented block should have a consistent indent (typically 4 spaces).**  
  In your file, some code blocks (for instance, the one in the macOS section) are indented by only 3 spaces instead of 4. This leads Sphinx to report “Unexpected indentation” errors.

Check all your code block directives and list items to ensure that after a directive you have:
1. A blank line.
2. A block indented with a consistent amount (commonly 4 spaces).

Fixing these indentation issues should resolve the warnings and errors during the documentation build.

acul71 commented on docs/contributing.rst:
```diff
@@ -29,40 +29,121 @@ On MacOS, you can install them using the following command:
+
+        cmake --version
+
+4. **Make (Optional)**
```
Why you state (Optional) for make how do you run the make commands without it?
```bash
 make
clean-build - remove build artifacts
clean-pyc - remove Python file artifacts
clean - run clean-build and clean-pyc
dist - build package and cat contents of the dist directory
lint - fix linting issues with pre-commit
test - run tests quickly with the default Python
docs - generate docs and open in browser (linux-docs for version on linux)
package-test - build package and install it in a venv for manual testing
notes - consume towncrier newsfragments and update release notes in docs - requires bump to be set
release - package and upload a release (does not run notes target) - requires bump to be set

```

paschal533 commented:
Hi @acul71 

Thanks for catching those reST errors! I fixed the indentation issues:
- Added blank lines after all `.. code::` directives.
- Standardized indents to 4 spaces (e.g., fixed macOS section from 3 to 4).
- Ran `sphinx-build -b html docs docs/_build` on Windows and confirmed no errors. Opened the HTML in my browser, and the Windows section looks good.

Pushed the corrected version to this PR.

acul71 commented:
@paschal533 
Hi Paschal. 
I can see that this test is failing:
```bash
FAILED tests/core/tools/timed_cache/test_timed_cache.py::test_readding_after_expiry
```
It has nothing to do with the [doc file](https://github.com/libp2p/py-libp2p/pull/560/commits/78add930dfee1410f1343557604cf6ec2a9a3dd0) of this PR, but I have a bad feeling that we're not done with the timed_cache errors.
We have to dig dipper, about this....

pacrob commented:
@paschal533 While you're expanding on this document, you could move the information from the **Dependencies** header down into the respective OS's sections. Then all setup instructions will be grouped by OS, should make it easier for new users.

paschal533 commented:
> @paschal533 While you're expanding on this document, you could move the information from the **Dependencies** header down into the respective OS's sections. Then all setup instructions will be grouped by OS, should make it easier for new users.

Okay Paul. I kinda miss this message. Let me work on it right away.

paschal533 commented:
Hi @pacrob, I’ve reorganized docs/contributing.rst by moving the dependency info from "Dependencies" into the OS-specific sections under "Setting the stage".

pacrob reviewed (approved):
lgtm!
